Source/Core/Absy.cs

```diff
@@ -1565,6 +1551,8 @@ public PredicateCmd CreateCandidateInvariant(Expr e, string tag = null)
   [ContractClass(typeof(DeclarationContracts))]
   public abstract class Declaration : Absy, ICarriesAttributes
   {
+    public virtual IEnumerable<Declaration> SelfAndChildren => Enumerable.Repeat(this, 1);
```

Comment:
I would have done `yield return this;`, but I now see that `Enumerable.Repeat(this, 1);` will emit less code. Nice.